pkg/commands/os.go

```diff
@@ -78,11 +76,40 @@ func (c *OSCommand) RunExecutable(cmd *exec.Cmd) error {
 
 // ExecutableFromString takes a string like `docker ps -a` and returns an executable command for it
 func (c *OSCommand) ExecutableFromString(commandStr string) *exec.Cmd {
+	fmt.Println(commandStr)
 	splitCmd := str.ToArgv(commandStr)
-	// c.Log.Info(splitCmd)
 	return c.command(splitCmd[0], splitCmd[1:]...)
 }
 
+func (c *OSCommand) NewShellCmd(commandStr string) *exec.Cmd {
+	if cmdHasShellInvokation(commandStr) {
+		return c.ExecutableFromString(commandStr)
+	}
+
+	quotedCommand := ""
+	// Windows does not seem to like quotes around the command
+	if c.Platform.os == "windows" {
+		quotedCommand = strings.NewReplacer(
+			"^", "^^",
+			"&", "^&",
+			"|", "^|",
+			"<", "^<",
+			">", "^>",
+			"%", "^%",
+		).Replace(commandStr)
+	} else {
+		quotedCommand = c.Quote(commandStr)
+	}
+
+	return c.ExecutableFromString(
+		fmt.Sprintf("%s %s %s", c.Platform.shell, c.Platform.shellArg, quotedCommand),
+	)
+}
+
+func cmdHasShellInvokation(cmd string) bool {
+	return strings.Contains(cmd, "bash -c") || strings.Contains(cmd, "sh -c")
+}
```

Comment:
Not sure if this covers all cases, but covers the most common shells. Covers:
- /bin/bash
- /bin/sh
- bash -c
- sh -c

Reply:
I'm not sure checking the string itself is a good approach: there's going to be edge cases we miss (e.g. custom shells like fish, zsh). I think we need to put this on the user somehow. We can add a `shell` key (boolean value) to the custom command config, which defaults to false. Unfortunately this means the user will need to actually set that to if they want to have it invoked from their shell, but that's the only way to ensure backwards compatibility right now.

Although I think 'true' would be a better default, I'm happy to make this change and then see if people complain and think it's worth breaking backwards compatibility to change the default

Reply:
Awesome, makes sense. I'll make that change.